omp private variable optimization episode 2

The next episode of #1259 

```
import numpy as np
#pythran export aaa(int):
def aaa(a):
    grid = np.empty((a, ))
    # omp parallel for private(i, j)
    for i in range(a):
        if i % 2:
            j = i * i
        else:
            j = i * i *3
        grid[i] = j

```

`/tmp/tmpSSj7KT.cpp:66:43: error: use of undeclared identifier 'j'`

--- a/pythran/backend.py
+++ b/pythran/backend.py
@@ -224,6 +224,7 @@ class CxxFunction(Backend):
         Not possible for function yielding values.
         """
         local_vars = self.scope[node].difference(skipped)
+        local_vars = local_vars.difference(self.openmp_deps)
         if not local_vars:
             return node_visited  # no processing
 
@@ -642,7 +643,8 @@ class CxxFunction(Backend):
                whole function.
         """
         auto_for = (isinstance(node.target, ast.Name) and
-                    node.target.id in self.scope[node])
+                    node.target.id in self.scope[node] and
+                    node.target.id not in self.openmp_deps)
         auto_for &= not metadata.get(node, OMPDirective)
         auto_for &= node.target.id not in self.openmp_deps
         return auto_for

--- a/pythran/tests/openmp.legacy/pythran_private.py
+++ b/pythran/tests/openmp.legacy/pythran_private.py
@@ -1,0 +1,15 @@
+
+import numpy as np
+def aaa(a):
+    grid = np.empty((a, ))
+    # omp parallel for private(i, j)
+    for i in range(a):
+        if i % 2:
+            j = i * i
+        else:
+            j = i * i *3
+        grid[i] = j
+
+def pythran_private():
+    from random import randint
+    return aaa(randint(5,5))

